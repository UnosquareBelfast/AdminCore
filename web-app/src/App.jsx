import React from 'react';
import Header from './components/Header';
import withAuth from './hoc/withAuth';
import { Dashboard } from './pages';

class App extends React.Component {
  render() {
    return (
      <div>
        <Header user={this.props.user} history={this.props.history} />
        <Dashboard user={this.props.user} />
      </div>
    );
  }
}

export default withAuth(App);
