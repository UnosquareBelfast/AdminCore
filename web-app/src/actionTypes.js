//USER
export const UPDATE_USER = 'UPDATE_USER';
export const RESET_USER = 'RESET_USER';

// DASHBOARD

export const SELECT_EVENT = 'SELECT_EVENT';
export const UPDATE_EVENT_DURATION = 'UPDATE_EVENT_DURATION';
export const TOGGLE_BOOKING_MODAL = 'TOGGLE_BOOKING_MODAL';
export const SET_IS_BEING_UPDATED = 'SET_IS_BEING_UPDATED';
export const SET_CALENDAR_EVENTS = 'SET_CALENDAR_EVENTS';
export const SET_ERROR = 'SET_ERROR';

// LOADING

export const SET_LOADING = 'SET_LOADING';
