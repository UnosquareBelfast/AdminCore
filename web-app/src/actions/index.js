export {
  fetchEvents,
  filterEventsByEmployeeId,
  updateBookingAndDuration,
  toggleBookingModal,
  updateBooking,
  updateBookingDuration,
} from './dashboard';
