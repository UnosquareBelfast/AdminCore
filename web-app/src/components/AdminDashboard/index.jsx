import React from 'react';
import { Link } from 'react-router-dom';
import { InnerLayout } from './styled';

const AdminDashboard = () => {
  return (
    <InnerLayout>
      <h2>Admin Dashboard</h2>
      <div className="columnWrap">
        <div>
          <h3>Employees</h3>
          <Link to="/admin/employees">View Employees</Link>
        </div>
        <div>
          <h3>Holidays</h3>
          <Link to="/admin/holidays">View All Holidays</Link>
          <Link to="/admin/pendingHolidays">View Pending Holidays</Link>
        </div>
        <div>
          <h3>Clients</h3>
          <Link to="">View All Holidays</Link>
          <Link to="">View Pending Holidays</Link>
        </div>
        <div>
          <h3>Contracts</h3>
          <Link to="">View All Holidays</Link>
          <Link to="">View Pending Holidays</Link>
        </div>
      </div>
    </InnerLayout>
  );
};

export default AdminDashboard;
