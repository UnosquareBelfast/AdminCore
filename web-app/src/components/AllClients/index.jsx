import React, { Fragment } from 'react';
import { PropTypes as PT } from 'prop-types';
import container from './container';
import { Table, Button } from '../common';
import { ActiveDot, CTA } from './styled';
import employeeStatus from '../../utilities/employeeStatus';

export const AllClients = ({
  clients,
  viewClient,
  archive,
  createNewClient,
}) => {
  let clientList;
  if (clients.length > 0) {
    clientList = clients.map(client => (
      <tr key={client.clientId}>
        <td>{client.clientName}</td>
        <td>{client.clientStatusDescription}</td>
        <td>
          <ActiveDot active={client.clientStatusId === employeeStatus.ACTIVE} />
          {client.clientStatusId === employeeStatus.ACTIVE
            ? 'Active'
            : 'Inactive'}
        </td>
        <td>
          <button onClick={() => viewClient(client.clientId)}>
            Edit Client
          </button>
          {client.clientStatusId === employeeStatus.ACTIVE ? (
            <button className="success" onClick={() => archive(client)}>
              Archive
            </button>
          ) : (
            <button className="error" onClick={() => archive(client, true)}>
              Activate
            </button>
          )}
        </td>
      </tr>
    ));
  } else {
    clientList = (
      <tr>
        <td>No Clients</td>
        <td />
        <td />
        <td />
      </tr>
    );
  }
  return (
    <Fragment>
      <h2>All Clients</h2>
      <Table
        tableHeaders={['Name', 'Description', 'Status', ' ']}
        tableRows={clientList}
      />
      <CTA>
        <Button onClick={() => createNewClient()} label="New Client" />
      </CTA>
    </Fragment>
  );
};

AllClients.propTypes = {
  createNewClient: PT.func,
  clients: PT.array,
  archive: PT.func.isRequired,
  viewClient: PT.func.isRequired,
};

export default container(AllClients);
