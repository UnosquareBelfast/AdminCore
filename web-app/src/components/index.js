export { default as AdminDashboard } from './AdminDashboard';
export { default as AllClients } from './AllClients';
export { default as AllHolidays } from './AllHolidays';
export { default as BigCalendarToolbar } from './BigCalendarToolbar';
export { default as BookedLeave } from './BookedLeave';
export { default as BookingCalendar } from './BookingCalendar';
export { default as BookingModal } from './BookingModal';
export { default as CreateClient } from './CreateClient';
export { default as CreateContract } from './CreateContract';
export { default as ClientModal } from './ClientModal';
export { default as CreateTeam } from './CreateTeam';
export { default as CreateUser } from './CreateUser';
export { default as Legend } from './Legend';
export { default as PendingHolidays } from './PendingHolidays';
export { default as RangeDayPicker } from './common/RangeDayPicker';
export { default as RequestHoliday } from './RequestHoliday';
export { default as TakenLeave } from './TakenLeave';
export { default as TeamView } from './TeamView';
export { default as UserDetails } from './UserDetails';
export { default as UserModal } from './UserModal';
export { default as ViewContracts } from './ViewContracts';
export { default as ViewTeams } from './ViewTeams';
export { default as HolidayModal } from './HolidayModal';
export { default as AllEmployees } from './AllEmployees';
export { default as DataTable } from './DataTable';
