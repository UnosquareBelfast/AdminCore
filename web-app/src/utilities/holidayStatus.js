import React from 'react';
import FontAwesomeIcon from '@fortawesome/react-fontawesome';
import {
  faThumbsUp,
  faThumbsDown,
  faClock,
  faSun,
} from '@fortawesome/fontawesome-free-solid';

export default {
  PENDING: 1,
  APPROVED: 2,
  REJECTED: 3,
  MANDATORY: 4,
};

export const statusText = [
  null,
  'Pending',
  'Approved',
  'Rejected',
  'Mandatory',
];

export const statusIcons = [
  null,
  <FontAwesomeIcon icon={faClock} />,
  <FontAwesomeIcon icon={faThumbsUp} />,
  <FontAwesomeIcon icon={faThumbsDown} />,
  <FontAwesomeIcon icon={faSun} />,
];
