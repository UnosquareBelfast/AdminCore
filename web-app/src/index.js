import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import App from './App';
import { Login } from './pages';

ReactDOM.render(
  <Router>
    <Switch>
      <Route exact path="/login" component={Login} />
      <Route path="/" component={App} />
    </Switch>
  </Router>,
  document.getElementById('app')
);
