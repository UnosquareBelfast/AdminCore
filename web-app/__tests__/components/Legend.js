import React from 'react';
import { shallow } from 'enzyme';
import { Legend } from '../../src/components';
import holidayStatus from '../../src/utilities/holidayStatus';

describe('Legend', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Legend />);

    expect(wrapper.exists());
    expect(wrapper.children().length).toEqual(1);
    expect(wrapper.childAt(0).childAt(0).text()).toEqual('Legend');
    expect(wrapper.children().childAt(1).prop('status')).toEqual(holidayStatus.PENDING);
    expect(wrapper.children().childAt(2).prop('status')).toEqual(holidayStatus.APPROVED);
    expect(wrapper.children().childAt(3).prop('status')).toEqual(holidayStatus.REJECTED);
    expect(wrapper.children().childAt(1).children().text()).toEqual('Pending');
    expect(wrapper.children().childAt(2).children().text()).toEqual('Approved');
    expect(wrapper.children().childAt(3).children().text()).toEqual('Rejected');
  });
});
