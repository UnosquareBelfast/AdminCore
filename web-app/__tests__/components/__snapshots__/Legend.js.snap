// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Legend snapshot 1`] = `
<Legend
  employees={
    Array [
      Object {
        "displayValue": "All",
        "value": -1,
      },
    ]
  }
  eventsKeyList={
    Array [
      Object {
        "active": false,
        "eventStatusId": 1,
        "key": 1,
        "type": "event",
      },
      Object {
        "active": false,
        "eventStatusId": 2,
        "key": 2,
        "type": "event",
      },
      Object {
        "active": false,
        "eventStatusId": 3,
        "key": 3,
        "type": "event",
      },
      Object {
        "active": false,
        "eventStatusId": 4,
        "key": 4,
        "type": "event",
      },
      Object {
        "active": false,
        "eventStatusId": 5,
        "key": 5,
        "type": "status",
      },
      Object {
        "active": false,
        "eventStatusId": 6,
        "key": 6,
        "type": "status",
      },
      Object {
        "active": false,
        "eventStatusId": 7,
        "key": 7,
        "type": "status",
      },
    ]
  }
  formStatus={[Function]}
  onToggleKey={[Function]}
  selectedEmployee={
    Object {
      "employeeId": -1,
    }
  }
/>
`;
